Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+- A DNS record dedicated to the blue and green clusters IP
+- A DNS record dedicated to the blue green cluster pointing to the App Gateway IP
+
+
+#### T0: Blue Cluster is On
+The initial stage of the pattern is to have the existing live cluster on, which is called the Blue Cluster. At this stage we are preparing for the deployment of the new version of the cluster.
+
+![Step0](../media/bg-step0.png)
+
+The trigger condition for the [next stage](#t1-green-cluster-deployment) is the release of a new version of the cluster.
+
+
+#### T1: Green Cluster Deployment
+Now the deployment of the new version is started and the the new Green Cluster is deployed in parallel to the existing Blue Cluster. After the Green cluster is deployed, the live traffic is still routed in the Blue Cluster. 
+
+![Step1](../media/bg-step1.png)
+
+The trigger to move into the [T2 stage](#t2-sync-k8s-state-between-blue-and-green-cluster) is the validation of AKS at platform level, using native metrics available in Azure Monitor and CLI commands to check the health of the deployment.
+
+
+#### T2: Sync K8S State between Blue and Green cluster
+At this stage there is an parity between the two clusters, meaning all **applications**, **operators** and **K8S resources** are newly deployed in the green cluster. The ultimate goal is that at the end of the sync the clusters are equivalent.
+
+There are multiple solutions/approaches to replicate/sync K8S state on clusters:
+- Redeployment via CI/CD
+- GitOps with solutions promoted in CNCF
+- Customized solution that store the K8S configs and resources, usually databases and K8S manifests generators
+
+![Step2](../media/bg-step2.png)
+
+Usually syncing the deployment of new applications is not permitted in the live cluster, so there is a period of time that starts with the sync and finishs when the switch to the green is completed. This period can be avoided with advanced mechanisms that manage the K8S state in multiple clusters, but that is not part of this article.
+
+When the sync is completed, a test/validation of the cluster is required. This include a check on the monitoring and logging platforms to validate the health of the cluster. Usually the Green cluster is exposed on the App Gateway or External LB with an internal URL, that is not visibile for external users.
+
+This test/validation stage is used as a trigger for the [next state](#t3-traffic-switch-to-the-green-cluster) of the pattern.
+
+
+#### T3: Traffic Switch to the green cluster
+After the sync is complete and Green cluster vaidated at platform and application level, then the Green cluster is ready to be promoted as the primary cluster to start receiving the live traffic. The switch is done at the networking level. Often the workloads are stateless. However, if the workloads are stateful then an additional solution must be implemented to mantain the state and caching during the switch.
```

Comment:
Regarding the last sentence in this paragraph., didn't you mention this already in the previous step T2?

Reply:
this is more related to the application itself and not the K8S state. I mean that if you have the concept of session or you have a stateful component like a database then is up to the application team define the proper strategy in order that blue green will not break from an end user perspective.
@bahramr